tests/python/pants_test/backend/python/tasks/test_interpreter_selection_integration.py

```diff
@@ -90,3 +91,36 @@ def _build_pex(self, binary_target, config=None, args=None, version=PY_27):
         ]
     command = ['binary', binary_target] + args
     return self.run_pants(command=command, config=config)
+
+  def test_stale_interpreter_purge_integration(self):
+    target = '{}:{}'.format(self.testproject, 'echo_interpreter_version')
+    config = {
+      'python-setup': {
+        'interpreter_constraints': ['CPython>=2.7,<4'],
+      }
+    }
+    with self.temporary_workdir() as workdir:
+      pants_run = self.run_pants_with_workdir(
+        ["run", target],
+        workdir=workdir,
+        config=config
+      )
+      self.assert_success(pants_run)
+
+      def _prepend_to_interpreter_path_file(path):
+        with open(path, 'r') as fp:
+          file_data = fp.readlines()
+          file_data[0] = '/my/bogus/interpreter/python2.7'
+        with open(path, 'w') as fp:
+          fp.writelines(file_data)
+
+      # Mangle interpreter.info.
+      for path in glob.glob(os.path.join(workdir, 'pyprep/interpreter/*/interpreter.info')):
+        _prepend_to_interpreter_path_file(path)
+
+      pants_run = self.run_pants_with_workdir(
+        ["run", target],
+        workdir=workdir,
+        config=config
+      )
+      self.assert_success(pants_run)
```

Comment:
Rather than just success, this could assert that the interpreter.infos no longer contain the bogus paths, or that the debug log was printed.

I think that might make it more clear what the expected outcome is.

Reply:
Agreed, I'll add the former.

Reply:
👍